packages/itwin/grouping-mapping-widget/src/widget/components/context/GroupingMappingCustomUIContext.ts

```diff
@@ -8,13 +8,18 @@ import type { GroupingMappingCustomUI } from "../customUI/GroupingMappingCustomU
 export const createGroupingMappingCustomUI = (
   customUIs: GroupingMappingCustomUI[] | undefined,
 ): GroupingMappingCustomUI[] => {
-  if (undefined === customUIs) {
-    return [];
-  }
-  return customUIs;
+  return customUIs ?? [];
 };
 
-export const GroupingMappingCustomUIContext = createContext<GroupingMappingCustomUI[]>([]);
+export interface IGroupingMappingCustomUI {
+  customUIs: GroupingMappingCustomUI[];
+  setCustomUIs: (customUI: GroupingMappingCustomUI[] | ((customUI: GroupingMappingCustomUI[]) => GroupingMappingCustomUI[])) => void;
+}
+
+export const GroupingMappingCustomUIContext = createContext<IGroupingMappingCustomUI>({
+  customUIs: [],
+  setCustomUIs: (customUI) => customUI,
```

Comment:
I don't understand this default context property value. Could you explain?

Reply:
It is just a no-op. A placeholder for a setter function.

Reply:
Why does a setter need to have a return value / type?

Reply:
Good point. Looking back at the type, it shouldn't return a value. I changed it.

Reply:
Update the types of those contexts as well. Now they accept a setter that does not return a value or returns the same type of value as the argument.

Reply:
I messed up with the previous comment. I meant this: `setShowGroupColor: (showGroupColor: boolean | ((showGroupColor: boolean) => boolean)) => void;` . Why does a setter accept a value or a function that accepts the same value and returns it?

Reply:
I made it stricter.